<?xml version="1.0"?>

<robot name="fixedwing" xmlns:xacro="http://ros.org/wiki/xacro">
  <!-- Properties -->
  <xacro:property name="namespace" value="fixedwing"/>
<xacro:property name="linkName" value="6fixedwing"/>
  <xacro:property name="wind_speed_topic" value="gazebo/wind_speed"/>
  <xacro:property name="command_topic" value="command"/>
  <xacro:property name="mesh_file" value="model://zephyr_delta_wing/meshes/wing.dae"/>
  <xacro:property name="body_width" value="1.54"/>
  <xacro:property name="body_height" value="0.77" />
  <xacro:property name="body_length" value="0.20"/>
  <xacro:property name="mass" value="1.5"/>
  <xacro:property name="Jx" value="0.1147"/>
  <xacro:property name="Jy" value="0.0576"/>
  <xacro:property name="Jz" value="0.1712"/>
  <xacro:property name="Jxz" value="0.0015"/>

  <!-- Property Blocks -->
  <xacro:property name="empty_inertia">
    <inertia ixx="0.0001" ixy="0.0" ixz="0.0" iyy="0.0001" iyz="0.0" izz="0.001" />
  </xacro:property>

  <!-- Instantiate Link and Visual -->
  <link name="${namespace}/base_link">
    <inertial>
      <mass value="${mass}" />
      <origin xyz="0 0 0" />
      <inertia ixx="${Jx}" ixy="0.00" ixz="${Jxz}" iyy="${Jy}" iyz="0.00" izz="${Jz}" />
    </inertial>
    <visual name="plane_visual">
      <origin xyz="0 0 0" rpy="0 0 1.570796" />
      <geometry>
        <mesh filename="${mesh_file}">
          <submesh name="Wing" center="true" />
        </mesh>
      </geometry>
    </visual>
    <collision name="plane_collision">
      <origin xyz="0 0 0" rpy="0 0 1.570796" />
      <geometry>
        <box size="${body_width} ${body_height} ${body_length}" />
      </geometry>
    </collision>
  </link>

  <gazebo>
    <plugin name="state_plugin" filename="libstate_plugin.so" >

        <namespace>fixedwing</namespace>
	<linkName>${namespace}/base_link</linkName>
 
      </plugin>
    </gazebo>
  <gazebo>
    <plugin name="Forces_Moments" filename="libForces_Moments.so" >

        <namespace>fixedwing</namespace>
	<linkName>${namespace}/base_link</linkName>
 
      </plugin>
    </gazebo>
  <gazebo>
    <plugin name="EOM" filename="libEOM.so" >

        <namespace>fixedwing</namespace>
	      <linkName>${namespace}/base_link</linkName>
 
      </plugin>
    </gazebo>




</robot>
